tailf.go

```diff
@@ -288,10 +298,93 @@ func (f *follower) updateFile() error {
 	}
 }
 
+// Note: if the file gets truncated, and before the size can be stat'd,
+// it has regrown to be >= the same size as previously, the truncate
+// will be missed. tl;dr, don't use copy-truncate...
+func (f *follower) checkForTruncate() error {
+	f.mu.Lock()
+
+	fi, err := os.Stat(f.filename)
+
+	f.mu.Unlock()
+	if os.IsNotExist(err) {
+		return ErrFileRemoved{fmt.Errorf("File was removed: %v", f.filename)}
+	}
+	if err != nil {
+		return err
+	}
+
+	newSize := fi.Size()
+	if newSize < f.size {
+		err = ErrFileTruncated{fmt.Errorf("File (%s) was truncated", f.filename)}
+	} else {
+		err = nil
+	}
+
+	f.size = newSize
+	return err
+}
+
+// This is here for situations where the directory the watched file sits in can't be inotified on
+func (f *follower) pollForChanges() {
+	previous_file, err := f.file.Stat()
+	if err != nil {
+		f.errc <- err
+	}
+
+	if err := f.watch.Add(f.filename); err != nil {
+		f.errc <- err
+	}
+
+	for current_file, err := os.Stat(f.filename); ; current_file, err = os.Stat(f.filename) {
```

Comment:
you could make this the first call of the loop:

```go
for {
    currentFile, err := os.Stat(f.filename)
}
```

also, no `snake_case` in Go =P